File: src/main/java/com/eadms/service/MarksService.java
package com.eadms.service;

import com.eadms.dto.request.MarksEntryRequest;
import com.eadms.dto.response.MarksResponse;

import java.util.List;

public interface MarksService {
    MarksResponse enterMarks(MarksEntryRequest request);
    MarksResponse updateMarks(Long marksId, MarksEntryRequest request);
    void deleteMarks(Long marksId);
    List<MarksResponse> getMarksByStudent(Long studentId);
    List<MarksResponse> getMarksByCourse(Long courseId);
    Double calculateAverageMarks(Long courseId);
    Double calculateStudentGPA(Long studentId);
}

File: frontend/src/services/marksService.ts
import api from './api'
import { Marks } from '@/types'

export const marksService = {
  async enterMarks(marks: Partial<Marks>): Promise<Marks> {
    const response = await api.post<Marks>('/teacher/marks', marks)
    return response.data
  },

  async updateMarks(id: number, marks: Partial<Marks>): Promise<Marks> {
    const response = await api.put<Marks>(`/teacher/marks/${id}`, marks)
    return response.data
  },

  async deleteMarks(id: number): Promise<void> {
    await api.delete(`/teacher/marks/${id}`)
  },

  async getMarksByStudent(studentId: number): Promise<Marks[]> {
    const response = await api.get<Marks[]>(`/teacher/marks/student/${studentId}`)
    return response.data
  },

  async getMarksByCourse(courseId: number): Promise<Marks[]> {
    const response = await api.get<Marks[]>(`/teacher/marks/course/${courseId}`)
    return response.data
  },
}
